streaming/consumer.cc

```diff
@@ -35,7 +35,7 @@ std::function<future<> (mutation_reader)> make_streaming_consumer(sstring origin
 
             auto cf = db.local().find_column_family(reader.schema()).shared_from_this();
             auto guard = service::topology_guard(frozen_guard);
-            auto use_view_update_path = co_await db::view::check_needs_view_update_path(vb.local(), db.local().get_token_metadata(), *cf, reason);
+            auto use_view_update_path = co_await db::view::check_needs_view_update_path(vb.local(), db.local().get_token_metadata_ptr(), *cf, reason);
```

Comment:
This caller could be made safe, if it held erm.

Should it not hold it? The answer to use_view_update_path() could change if it doesn't.

Reply:
That's out of scope of this patch that only fixes the use-after-free.
As for changing the algorithm, we should consider it as a follow up, if it's indeed needed.